### PR TITLE
Several fixes after 1st production test

### PR DESCRIPTION
**Pull request by egabancho**

### CODE REVIEW

**Comment by egabancho** [comment]
Hey @tiborsimko @drjova!
since you've shown some interest about using a S3 like file service in the future, would you mind taking a look at the changes I did?
I know that it might be a bit puzzling, but I just need to know if I am doing something stupid
All the changes are tested already with a production ready service, so they should work 😄

File: invenio_s3/__init__.py [hunk 1]
```diff
@@ -16,8 +16,28 @@
 <https://invenio-files-rest.readthedocs.io/en/latest/api.html#module-invenio_files_rest.models>`_
 for each S3 bucket. This is just for simplicity, it can used however needed.
 
-This module doesn't create S3 buckets automatically, so before starting they
-need to be created.
+When creating a new location which will use the S3 API, the URI needs to start
+with ``s3://``, for example
+``invenio files location S3_default s3://myserver.com/b1 --defauilt`` will
```
Comment (jirikuncar) [inline invenio_s3/__init__.py]:
```suggestion
``invenio files location S3_default s3://myserver.com/b1 --default`` will
```

File: invenio_s3/config.py [hunk 2]
```diff
@@ -38,3 +38,11 @@
 <https://boto3.readthedocs.io/en/latest/guide/configuration.html#credentials>`_.
 for more information.
 """
+
+S3_URL_EXPIRATION = 60
+""""Number of seconds the file serving URL will be valid.
```
Comment (jirikuncar) [inline invenio_s3/config.py]:
```suggestion
"""Number of seconds the file serving URL will be valid.
```

File: invenio_s3/ext.py [hunk 3]
```diff
@@ -30,6 +30,12 @@ def init_s3f3_info(self):
             key=current_app.config.get('S3_ACCCESS_KEY_ID', ''),
             secret=current_app.config.get('S3_SECRECT_ACCESS_KEY', ''),
             client_kwargs={},
+            config_kwargs={
+                's3': {
+                    'addressing_style': 'path'
```
Comment (jirikuncar) [inline invenio_s3/ext.py]:
```suggestion
                    'addressing_style': 'path',
```

File: invenio_s3/ext.py [hunk 4]
```diff
@@ -30,6 +30,12 @@ def init_s3f3_info(self):
...
+                },
+                'signature_version': 's3v4'
```
Comment (jirikuncar) [inline invenio_s3/ext.py]:
```suggestion
                'signature_version': 's3v4',
```

**Comment by egabancho** [comment]
Thanks for the feedback @jirikuncar !

**Comment by drjova** [comment]
@egabancho lgtm 🥇

**Pull request merged**